### PR TITLE
[fix](multi-catalog)fix getting ugi methods and unify them

### PR DESCRIPTION
**Pull request by wsjz**
## Proposed changes

put all ugi login methods to HadoopUGI

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by wsjz** [comment]
run buildall

File: fe/fe-common/src/main/java/org/apache/doris/common/security/authentication/HadoopUGI.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+                    // if the current user is logged by kerberos and is the same user
+                    // just use checkTGTAndReloginFromKeytab because this method will only relogin
+                    // when the TGT is expired or is close to expiry
+                    ugi.checkTGTAndReloginFromKeytab();
+                    return ugi;
+                }
+            } catch (IOException e) {
+                LOG.warn("A SecurityException occurs with kerberos, do login immediately.", e);
+            }
+            String keytab = conf.get(HADOOP_KERBEROS_KEYTAB);
```
Comment (morningman) [inline fe/fe-common/src/main/java/org/apache/doris/common/security/authentication/HadoopUGI.java:62]: Some place may required other principal and keytab, eg, hive metastore client.
HADOOP_KERBEROS_KEYTAB and HADOOP_KERBEROS_PRINCIPAL is only for hdfs

Comment (wsjz) [reply fe/fe-common/src/main/java/org/apache/doris/common/security/authentication/HadoopUGI.java]: OK, add hive krb properties

File: fe/fe-common/src/main/java/org/apache/doris/common/security/authentication/HadoopUGI.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        }
+        return HadoopUGI.loginWithUGI(conf);
+    }
+
+    /**
+     * use for HMSExternalCatalog to login
+     * @param catalogProperty our properties
+     * @param hiveConf hive conf
+     */
+    public static void tryKrbLogin(String catalogName, Map<String, String> catalogProperty, Configuration hiveConf) {
```
[the remark this answers is not in the document]
Comment (wsjz) [reply fe/fe-common/src/main/java/org/apache/doris/common/security/authentication/HadoopUGI.java]: not change

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run buildall

File: fe/fe-common/src/main/java/org/apache/doris/common/security/authentication/HadoopUGI.java [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+                                                         String krbPrincipalKey,
+                                                         String krbKeytabKey) {
+        AuthenticationConfig config = new AuthenticationConfig();
+        String authentication = conf.get(HADOOP_SECURITY_AUTHENTICATION, null);
+        if (AuthType.KERBEROS.getDesc().equals(authentication)) {
+            config.setKerberos(true);
+            config.setKerberosPrincipal(conf.get(krbPrincipalKey));
+            config.setKerberosKeytab(conf.get(krbKeytabKey));
+            conf.set(HADOOP_KERBEROS_AUTHORIZATION, "true");
+            UserGroupInformation.setConfiguration(conf);
```
Comment (morningman) [inline fe/fe-common/src/main/java/org/apache/doris/common/security/authentication/HadoopUGI.java:65]: Why adding this line?
This should be done when doing `loginWithUGI`

File: fe/fe-common/src/main/java/org/apache/doris/common/security/authentication/HadoopUGI.java [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.IOException;
+
+public class HadoopUGI {
+
+    private static final Logger LOG = LogManager.getLogger(HadoopUGI.class);
+    public static String HADOOP_USER_NAME = "hadoop.username";
+    public static String HADOOP_SECURITY_AUTHENTICATION = "hadoop.security.authentication";
```
Comment (morningman) [inline fe/fe-common/src/main/java/org/apache/doris/common/security/authentication/HadoopUGI.java:32]: Some of these name have been defined in HdfsResource.java

Comment (wsjz) [reply fe/fe-common/src/main/java/org/apache/doris/common/security/authentication/HadoopUGI.java:32]: HdfsResource is not in fe-common, be-java-extensions cannot use it.

File: fe/fe-common/src/main/java/org/apache/doris/common/security/authentication/HadoopUGI.java (hunk 3 above)
Comment (wsjz) [reply fe/fe-common/src/main/java/org/apache/doris/common/security/authentication/HadoopUGI.java:65]: loginWithUGI has removed conf param

**Review by morningman — APPROVED** [review]
LGTM

**Comment by wsjz** [comment]
run buildall

**Review by zddr — APPROVED** [review]

**Pull request merged**